### ISSUE TITLE
Listing vox environments takes a long time

### ISSUE THREAD

**Issue by rnovacek** [xonsh/xonsh#2785]
Since version `0.7.4` of xonsh, listing vox environments is significantly slower that it used to be. It esp. painful for me because I use `avox` to automatically activate the venv when I switch directories. So now navigation around in filesystem takes a lot of time.

## xonfig
```
+------------------+----------------------+
| xonsh            | 0.7.6                |
| Git SHA          | 3939a31b             |
| Commit Date      | Aug 21 16:03:45 2018 |
| Python           | 3.6.6                |
| PLY              | 3.9                  |
| have readline    | True                 |
| prompt toolkit   | 2.0.3                |
| shell type       | prompt_toolkit2      |
| pygments         | 2.2.0                |
| on posix         | True                 |
| on linux         | True                 |
| distro           | fedora               |
| on darwin        | False                |
| on windows       | False                |
| on cygwin        | False                |
| on msys2         | False                |
| is superuser     | False                |
| default encoding | utf-8                |
| xonsh encoding   | utf-8                |
| encoding errors  | surrogateescape      |
+------------------+----------------------+
```

## Expected Behavior
`vox ls` should finish immediately.

## Current Behavior
`vox ls` take 2 - 3 seconds, rendering `avox` unusable.

## Steps to Reproduce
Create a couple of big virtual environments in `~/.virtualenvs` and run `vox ls`.

## Details
I think it boils down to this PR:
https://github.com/xonsh/xonsh/pull/2771

According to the docs, `dirnames.remove` will skip iteration through subdirectories, skipping the rest of the loop with `continue` won't skip those subdirs, so it has to go through entire directory tree.

**Comment by scopatz** [comment]
Thanks for reporting this @rnovacek! I think you are right that this is because of #2771.  I don't use vox / avox, so is there a more complete example that you can give?  It seems to me that the filters in #2771 must not work correctly.

**Comment by rnovacek** [comment]
You can create some virtual environments like this:

```python
xontrib load vox
for i in range(10):
    vox new @(f'test{i}')
    vox activate @(f'test{i}')
    pip3 install django
    vox deactivate
```

Then running `vox ls` will take non-trivial amount of time (1.5 second for me).

**Comment by gsaga** [comment]
@scopatz What errors does #2771 fix. I'm not getting any errors on running `vox list` with commit 650524 reverted, and this tutorial mentions that dirnames has to be modified in place(near the end of the article):
https://www.pythoncentral.io/how-to-traverse-a-directory-tree-in-python-guide-to-os-walk/
This contradicts the commit message of the above commit.

**Comment by scopatz** [comment]
merged #2788 - closing!
